diff -ru perl5.003_07.orig/config_h.SH perl5.003_07/config_h.SH
--- perl5.003_07.orig/config_h.SH	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/config_h.SH	2008-09-19 17:53:28.000000000 +0100
@@ -563,6 +563,30 @@
  */
 #$d_sem HAS_SEM		/**/
 
+/* HAS_UNION_SEMUN:
+ *	This symbol, if defined, indicates that the union semun
+ *	is defined in <sys/sem.h>.  If not, the user code probably
+ *      needs to define it as:
+ * union semun {
+ *    int val;
+ *    struct semid_ds *buf;
+ *    unsigned short *array;
+ * }
+ */
+#$d_union_semun HAS_UNION_SEMUN		/**/
+
+/* USE_SEMCTL_SEMUN:
+ *	This symbol, if defined, indicates that union semun is
+ *	used for semctl IPC_STAT.
+ */
+#$d_semctl_semun USE_SEMCTL_SEMUN	/**/
+
+/* USE_SEMCTL_SEMID_DS:
+ *	This symbol, if defined, indicates that struct semid_ds * is
+ *	used for semctl IPC_STAT.
+ */
+#$d_semctl_semid_ds USE_SEMCTL_SEMID_DS	/**/
+
 /* HAS_SETEGID:
  *	This symbol, if defined, indicates that the setegid routine is available
  *	to change the effective gid of the current program.
diff -ru perl5.003_07.orig/Configure perl5.003_07/Configure
--- perl5.003_07.orig/Configure	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/Configure	2008-09-19 18:09:58.000000000 +0100
@@ -194,6 +194,7 @@
 zip=''
 full_sed=''
 libswanted=''
+ignore_versioned_solibs=''
 hint=''
 myuname=''
 osname=''
@@ -343,6 +344,9 @@
 d_semctl=''
 d_semget=''
 d_semop=''
+d_union_semun=''
+d_semctl_semun=''
+d_semctl_semid_ds=''
 d_setegid=''
 d_seteuid=''
 d_setlinebuf=''
@@ -506,6 +510,7 @@
 loclibpth=''
 plibpth=''
 xlibpth=''
+ignore_versioned_solibs=''
 libs=''
 lns=''
 lseektype=''
@@ -3240,7 +3245,8 @@
 esac
 for thislib in $libswanted; do
 	
-	if xxx=`./loc lib$thislib.$so.[0-9]'*' X $libpth`; $test -f "$xxx"; then
+	if xxx=`./loc lib$thislib.$so.[0-9]'*' X $libpth`;
+		$test -f "$xxx" -a "X$ignore_versioned_solibs" = "X"; then
 		echo "Found -l$thislib (shared)."
 		case " $dflt " in
 		*"-l$thislib "*);;
@@ -7056,6 +7062,162 @@
 set d_sem
 eval $setvar
 
+: see how to do semctl IPC_STAT
+case "$h_sem$d_sem$d_semctl" in
+true$define$define)
+    : see whether sys/sem.h defines union semun
+    $cat > try.c <<'END'
+#include <sys/types.h>
+#include <sys/ipc.h>
+#include <sys/sem.h>
+int main () { union semun semun; semun.buf = 0; }
+END
+    if $cc $ccflags -o try try.c > /dev/null 2>&1; then
+	echo "You have union semun in <sys/sem.h>." >&4
+	val="$define"
+    else
+	 echo "You do not have union semun in <sys/sem.h>." >&4
+	 val="$undef"
+    fi
+    $rm -f try try.c
+    set d_union_semun
+    eval $setvar
+
+    : see whether semctl IPC_STAT can use union semun
+    $cat > try.c <<END
+#include <sys/types.h>
+#include <sys/ipc.h>
+#include <sys/sem.h>
+#include <sys/stat.h>
+#include <stdio.h>
+#include <errno.h>
+#ifndef errno
+extern int errno;
+#endif
+#$d_union_semun HAS_UNION_SEMUN
+int main() {
+  union semun
+#ifndef HAS_UNION_SEMUN
+  {
+    int val;
+    struct semid_ds *buf;
+    unsigned short *array;
+  }
+#endif
+    arg;
+  int sem, st;
+
+#if defined(IPC_PRIVATE) && defined(S_IRWXU) && defined(S_IRWXG) && \
+    defined(S_IRWXO) && defined(IPC_CREAT)
+  sem = semget(IPC_PRIVATE, 1, S_IRWXU|S_IRWXG|S_IRWXO|IPC_CREAT);
+  if (sem > -1) {
+    struct semid_ds argbuf;
+    arg.buf = &argbuf;
+#   ifdef IPC_STAT
+    st = semctl(sem, 0, IPC_STAT, arg);
+    if (st == 0)
+      printf("semun\n");
+    else
+#   endif /* IPC_STAT */
+      printf("semctl IPC_STAT failed: errno = %d\n", errno);
+#   ifdef IPC_RMID
+    if (semctl(sem, 0, IPC_RMID, arg) != 0)
+#   endif /* IPC_RMID */
+      printf("semctl IPC_RMID failed: errno = %d\n", errno);
+  } else
+#endif /* IPC_PRIVATE && ... */
+    printf("semget failed: errno = %d\n", errno);
+
+  return 0;
+}
+END
+    val="$undef"
+    if $cc $ccflags -o try try.c > /dev/null 2>&1; then
+        d_semctl_semun=`./try`
+        case "$d_semctl_semun" in
+        semun) val="$define" ;;
+        esac
+    fi
+    $rm -f try try.c
+    set d_semctl_semun
+    eval $setvar
+    case "$d_semctl_semun" in
+    $define|true)
+        echo "You can use union semun for semctl IPC_STAT." >&4
+        ;;
+    *)  echo "You cannot use union semun for semctl IPC_STAT." >&4
+        ;;
+    esac
+
+    : see whether semctl IPC_STAT can use struct semid_ds pointer
+
+    $cat > try.c <<'END'
+#include <sys/types.h>
+#include <sys/ipc.h>
+#include <sys/sem.h>
+#include <sys/stat.h>
+#include <stdio.h>
+#include <errno.h>
+#ifndef errno
+extern int errno;
+#endif
+int main() {
+  struct semid_ds arg;
+  int sem, st;
+
+#if defined(IPC_PRIVATE) && defined(S_IRWXU) && defined(S_IRWXG) && \
+    defined(S_IRWXO) && defined(IPC_CREAT)
+  sem = semget(IPC_PRIVATE, 1, S_IRWXU|S_IRWXG|S_IRWXO|IPC_CREAT);
+  if (sem > -1) {
+#   ifdef IPC_STAT
+    st = semctl(sem, 0, IPC_STAT, &arg);
+    if (st == 0)
+      printf("semid_ds\n");
+    else
+#   endif /* IPC_STAT */
+      printf("semctl IPC_STAT failed: errno = %d\n", errno);
+#   ifdef IPC_RMID
+    if (semctl(sem, 0, IPC_RMID, &arg) != 0)
+#   endif /* IPC_RMID */
+      printf("semctl IPC_RMID failed: errno = %d\n", errno);
+  } else
+#endif /* IPC_PRIVATE && ... */
+      printf("semget failed: errno = %d\n", errno);
+
+  return 0;
+}
+END
+    val="$undef"
+    if $cc $ccflags -o try try.c > /dev/null 2>&1; then
+        d_semctl_semid_ds=`./try`
+        case "$d_semctl_semid_ds" in
+        semid_ds) val="$define" ;;
+        esac
+    fi
+    $rm -f try try.c
+    set d_semctl_semid_ds
+    eval $setvar
+    case "$d_semctl_semid_ds" in
+    $define|true)
+        echo "You can use struct semid_ds * for semctl IPC_STAT." >&4
+        ;;
+    *)  echo "You cannot use struct semid_ds * for semctl IPC_STAT." >&4
+        ;;
+    esac
+    ;;
+*)  val="$undef"
+
+    set d_union_semun
+    eval $setvar
+
+    set d_semctl_semun
+    eval $setvar
+
+    set d_semctl_semid_ds
+    eval $setvar
+    ;;
+esac
+
 : see if setegid exists
 set setegid d_setegid
 eval $inlibc
@@ -8790,7 +8952,7 @@
 $startsh
 $test -s signal.lst && exit 0
 if $cc $ccflags signal.c -o signal $ldflags >/dev/null 2>&1; then
-	./signal | $sort -n +1 | $uniq | $awk -f signal.awk >signal.lst
+   	./signal | ($sort -n -k 2 2>/dev/null || $sort -n +1) | $uniq | $awk -f signal.awk >signal.lst
 else
 	echo "(I can't seem be able to compile the test program -- Guessing)"
 	echo 'kill -l' >signal
@@ -9809,6 +9971,9 @@
 d_semctl='$d_semctl'
 d_semget='$d_semget'
 d_semop='$d_semop'
+d_union_semun='$d_union_semun'
+d_semctl_semun='$d_semctl_semun'
+d_semctl_semid_ds='$d_semctl_semid_ds'
 d_setegid='$d_setegid'
 d_seteuid='$d_seteuid'
 d_setlinebuf='$d_setlinebuf'
@@ -9961,6 +10126,7 @@
 i_varargs='$i_varargs'
 i_varhdr='$i_varhdr'
 i_vfork='$i_vfork'
+ignore_versioned_solibs='$ignore_versioned_solibs'
 incpath='$incpath'
 inews='$inews'
 installarchlib='$installarchlib'
diff -ru perl5.003_07.orig/doio.c perl5.003_07/doio.c
--- perl5.003_07.orig/doio.c	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/doio.c	2008-09-19 17:53:28.000000000 +0100
@@ -1383,12 +1383,13 @@
 	else if (cmd == GETALL || cmd == SETALL)
 	{
 	    struct semid_ds semds;
-	    if (semctl(id, 0, IPC_STAT, &semds) == -1)
+	    union semun semun;
+
+            semun.buf = &semds;
+	    if (Semctl(id, 0, IPC_STAT, semun) == -1)
 		return -1;
 	    getinfo = (cmd == GETALL);
-	    infosize = semds.sem_nsems * sizeof(short);
-		/* "short" is technically wrong but much more portable
-		   than guessing about u_?short(_t)? */
+	    infosize = semds.sem_nsems * sizeof(unsigned short);
 	}
 	break;
 #endif
@@ -1428,8 +1429,12 @@
 	break;
 #endif
 #ifdef HAS_SEM
-    case OP_SEMCTL:
-	ret = semctl(id, n, cmd, (struct semid_ds *)a);
+    case OP_SEMCTL: {
+            union semun unsemds;
+
+            unsemds.buf = (struct semid_ds *)a;
+	    ret = Semctl(id, n, cmd, unsemds);
+        }
 	break;
 #endif
 #ifdef HAS_SHM
diff -ru perl5.003_07.orig/hints/3b1.sh perl5.003_07/hints/3b1.sh
--- perl5.003_07.orig/hints/3b1.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/3b1.sh	2008-09-19 18:23:00.000000000 +0100
@@ -13,3 +13,4 @@
 echo "\nIf you want to use the 3b1 shared libraries, complete this script then"
 echo "read the header in 3b1cc.           [Type carriage return to continue]\c"
 read vch
+i_db=$undef
diff -ru perl5.003_07.orig/hints/aix.sh perl5.003_07/hints/aix.sh
--- perl5.003_07.orig/hints/aix.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/aix.sh	2008-09-19 18:23:00.000000000 +0100
@@ -69,3 +69,4 @@
 
 ;;
 esac
+i_db=$undef
diff -ru perl5.003_07.orig/hints/altos486.sh perl5.003_07/hints/altos486.sh
--- perl5.003_07.orig/hints/altos486.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/altos486.sh	2008-09-19 18:23:00.000000000 +0100
@@ -1,3 +1,4 @@
 : have heard of problems with -lc_s on Altos 486
 set `echo " $libswanted " | sed "s/ c_s / /"`
 libswanted="$*"
+i_db=$undef
diff -ru perl5.003_07.orig/hints/apollo.sh perl5.003_07/hints/apollo.sh
--- perl5.003_07.orig/hints/apollo.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/apollo.sh	2008-09-19 18:23:00.000000000 +0100
@@ -49,3 +49,4 @@
 #syntax errors as converted by h2ph. 
 
 # Generally, h2ph might need a lot of help.
+i_db=$undef
diff -ru perl5.003_07.orig/hints/aux.sh perl5.003_07/hints/aux.sh
--- perl5.003_07.orig/hints/aux.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/aux.sh	2008-09-19 18:23:00.000000000 +0100
@@ -18,3 +18,4 @@
 	echo "./Configure -Dcc=gcc"
 	;;
 esac
+i_db=$undef
diff -ru perl5.003_07.orig/hints/bsdos.sh perl5.003_07/hints/bsdos.sh
--- perl5.003_07.orig/hints/bsdos.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/bsdos.sh	2008-09-19 18:23:00.000000000 +0100
@@ -97,3 +97,4 @@
 	d_setruid='undef'
 	;;
 esac
+i_db=$undef
diff -ru perl5.003_07.orig/hints/convexos.sh perl5.003_07/hints/convexos.sh
--- perl5.003_07.orig/hints/convexos.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/convexos.sh	2008-09-19 18:23:00.000000000 +0100
@@ -10,3 +10,4 @@
 set X $myuname
 shift
 osvers=$4
+i_db=$undef
diff -ru perl5.003_07.orig/hints/cxux.sh perl5.003_07/hints/cxux.sh
--- perl5.003_07.orig/hints/cxux.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/cxux.sh	2008-09-19 18:23:00.000000000 +0100
@@ -99,3 +99,4 @@
 the tests, you can fool it into thinking it worked.
 
 EOM
+i_db=$undef
diff -ru perl5.003_07.orig/hints/dec_osf.sh perl5.003_07/hints/dec_osf.sh
--- perl5.003_07.orig/hints/dec_osf.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/dec_osf.sh	2008-09-19 18:23:00.000000000 +0100
@@ -9,3 +9,4 @@
 esac
 ccflags="$ccflags -DSTANDARD_C"
 lddlflags='-shared -expect_unresolved "*" -s -hidden'
+i_db=$undef
diff -ru perl5.003_07.orig/hints/dgux.sh perl5.003_07/hints/dgux.sh
--- perl5.003_07.orig/hints/dgux.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/dgux.sh	2008-09-19 18:23:00.000000000 +0100
@@ -112,3 +112,4 @@
 # emulation.
 ccflags="$ccflags -Dflock=dg_flock"
 d_flock=define
+i_db=$undef
diff -ru perl5.003_07.orig/hints/dynixptx.sh perl5.003_07/hints/dynixptx.sh
--- perl5.003_07.orig/hints/dynixptx.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/dynixptx.sh	2008-09-19 18:23:00.000000000 +0100
@@ -15,3 +15,4 @@
 # Configure defaults to usenm='y', which doesn't work very well
 usenm='n'
 
+i_db=$undef
diff -ru perl5.003_07.orig/hints/dynix.sh perl5.003_07/hints/dynix.sh
--- perl5.003_07.orig/hints/dynix.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/dynix.sh	2008-09-19 18:23:00.000000000 +0100
@@ -5,3 +5,4 @@
 # Reported by Craig Milo Rogers <Rogers@ISI.EDU>
 # Date: Tue, 30 Jan 96 15:29:26 PST
 d_casti32=undef
+i_db=$undef
diff -ru perl5.003_07.orig/hints/epix.sh perl5.003_07/hints/epix.sh
--- perl5.003_07.orig/hints/epix.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/epix.sh	2008-09-19 18:23:00.000000000 +0100
@@ -73,3 +73,4 @@
 before running make.
 
 EOM
+i_db=$undef
diff -ru perl5.003_07.orig/hints/esix4.sh perl5.003_07/hints/esix4.sh
--- perl5.003_07.orig/hints/esix4.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/esix4.sh	2008-09-19 18:23:00.000000000 +0100
@@ -39,3 +39,4 @@
 before running make.
 
 EOM
+i_db=$undef
diff -ru perl5.003_07.orig/hints/fps.sh perl5.003_07/hints/fps.sh
--- perl5.003_07.orig/hints/fps.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/fps.sh	2008-09-19 18:23:00.000000000 +0100
@@ -1 +1,2 @@
 ccflags="$ccflags -J"
+i_db=$undef
diff -ru perl5.003_07.orig/hints/freebsd.sh perl5.003_07/hints/freebsd.sh
--- perl5.003_07.orig/hints/freebsd.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/freebsd.sh	2008-09-19 18:23:00.000000000 +0100
@@ -79,3 +79,4 @@
 
 EOM
 
+i_db=$undef
diff -ru perl5.003_07.orig/hints/genix.sh perl5.003_07/hints/genix.sh
--- perl5.003_07.orig/hints/genix.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/genix.sh	2008-09-19 18:23:00.000000000 +0100
@@ -1 +1,2 @@
 i_varargs=undef
+i_db=$undef
diff -ru perl5.003_07.orig/hints/greenhills.sh perl5.003_07/hints/greenhills.sh
--- perl5.003_07.orig/hints/greenhills.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/greenhills.sh	2008-09-19 18:23:00.000000000 +0100
@@ -1 +1,2 @@
 ccflags="$ccflags -X18"
+i_db=$undef
diff -ru perl5.003_07.orig/hints/hpux.sh perl5.003_07/hints/hpux.sh
--- perl5.003_07.orig/hints/hpux.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/hpux.sh	2008-09-19 18:23:00.000000000 +0100
@@ -123,3 +123,4 @@
 #     assembler of the form:
 #          (warning) Use of GR3 when frame >= 8192 may cause conflict.
 #     These warnings are harmless and can be safely ignored.
+i_db=$undef
diff -ru perl5.003_07.orig/hints/i386.sh perl5.003_07/hints/i386.sh
--- perl5.003_07.orig/hints/i386.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/i386.sh	2008-09-19 18:23:00.000000000 +0100
@@ -1 +1,2 @@
 ldflags='-L/usr/ucblib'
+i_db=$undef
diff -ru perl5.003_07.orig/hints/irix_4.sh perl5.003_07/hints/irix_4.sh
--- perl5.003_07.orig/hints/irix_4.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/irix_4.sh	2008-09-19 18:23:00.000000000 +0100
@@ -22,3 +22,4 @@
 	-DSTANDARD_C -cckr 
 in ccflags.
 EOM
+i_db=$undef
diff -ru perl5.003_07.orig/hints/irix_5.sh perl5.003_07/hints/irix_5.sh
--- perl5.003_07.orig/hints/irix_5.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/irix_5.sh	2008-09-19 18:23:00.000000000 +0100
@@ -32,3 +32,4 @@
 # patchSG0000596.  The patch can be downloaded from Advantage OnLine (SGI's
 # WWW server) or from the Support Advantage 9/95 Patch CDROM.  Thanks to Tom 
 # Christiansen and others who provided assistance.
+i_db=$undef
diff -ru perl5.003_07.orig/hints/irix_6_2.sh perl5.003_07/hints/irix_6_2.sh
--- perl5.003_07.orig/hints/irix_6_2.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/irix_6_2.sh	2008-09-19 18:23:00.000000000 +0100
@@ -52,3 +52,4 @@
 shift
 libswanted="$*"
 
+i_db=$undef
diff -ru perl5.003_07.orig/hints/irix_6.sh perl5.003_07/hints/irix_6.sh
--- perl5.003_07.orig/hints/irix_6.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/irix_6.sh	2008-09-19 18:23:00.000000000 +0100
@@ -41,3 +41,4 @@
 # set `echo X "$libswanted "|sed -e 's/ socket / /' -e 's/ sun / /' -e 's/ crypt / /' -e 's/ nsl / /' -e 's/ dl / /'`
 # shift
 # libswanted="$*"
+i_db=$undef
diff -ru perl5.003_07.orig/hints/isc_2.sh perl5.003_07/hints/isc_2.sh
--- perl5.003_07.orig/hints/isc_2.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/isc_2.sh	2008-09-19 18:23:00.000000000 +0100
@@ -20,3 +20,4 @@
 # Compensate for conflicts in <net/errno.h>
 doio_cflags='ccflags="$ccflags -DENOTSOCK=103"'
 pp_sys_cflags='ccflags="$ccflags -DENOTSOCK=103"'
+i_db=$undef
diff -ru perl5.003_07.orig/hints/isc.sh perl5.003_07/hints/isc.sh
--- perl5.003_07.orig/hints/isc.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/isc.sh	2008-09-19 18:23:00.000000000 +0100
@@ -33,3 +33,4 @@
 # compilation "redefinition" warnings, but they appear harmless.
 # ccflags="$ccflags -D_SYSV3"
 
+i_db=$undef
diff -ru perl5.003_07.orig/hints/linux.sh perl5.003_07/hints/linux.sh
--- perl5.003_07.orig/hints/linux.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/linux.sh	2008-09-19 18:23:00.000000000 +0100
@@ -15,6 +15,27 @@
 # Updated Fri Jun 21 11:07:54 EDT 1996
 # NDBM support for ELF renabled by <kjahds@kjahds.com>
 
+# Debian and Red Hat, and perhaps other vendors, provide both runtime and
+# development packages for some libraries.  The runtime packages contain shared
+# libraries with version information in their names (e.g., libgdbm.so.1.7.3);
+# the development packages supplement this with versionless shared libraries
+# (e.g., libgdbm.so).
+#
+# If you want to link against such a library, you must install the development
+# version of the package.
+#
+# These packages use a -dev naming convention in both Debian and Red Hat:
+#   libgdbmg1  (non-development version of GNU libc 2-linked GDBM library)
+#   libgdbmg1-dev (development version of GNU libc 2-linked GDBM library)
+# So make sure that for any libraries you wish to link Perl with under
+# Debian or Red Hat you have the -dev packages installed.
+#
+# Some operating systems (e.g., Solaris 2.6) will link to a versioned shared
+# library implicitly.  For example, on Solaris, `ld foo.o -lgdbm' will find an
+# appropriate version of libgdbm, if one is available; Linux, however, doesn't
+# do the implicit mapping.
+ignore_versioned_solibs='y'
+
 # perl goes into the /usr tree.  See the Filesystem Standard
 # available via anonymous FTP at tsx-11.mit.edu in
 # /pub/linux/docs/linux-standards/fsstnd.
@@ -149,3 +170,4 @@
 fi
 
 
+i_db=$undef
diff -ru perl5.003_07.orig/hints/machten_2.sh perl5.003_07/hints/machten_2.sh
--- perl5.003_07.orig/hints/machten_2.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/machten_2.sh	2008-09-19 18:23:00.000000000 +0100
@@ -55,3 +55,4 @@
 Read the File::Find documentation for more information.
 
 EOM
+i_db=$undef
diff -ru perl5.003_07.orig/hints/mips.sh perl5.003_07/hints/mips.sh
--- perl5.003_07.orig/hints/mips.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/mips.sh	2008-09-19 18:23:00.000000000 +0100
@@ -12,3 +12,4 @@
 END
 ;;
 esac
+i_db=$undef
diff -ru perl5.003_07.orig/hints/mpc.sh perl5.003_07/hints/mpc.sh
--- perl5.003_07.orig/hints/mpc.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/mpc.sh	2008-09-19 18:23:00.000000000 +0100
@@ -1 +1,2 @@
 ccflags="$ccflags -X18"
+i_db=$undef
diff -ru perl5.003_07.orig/hints/mpeix.sh perl5.003_07/hints/mpeix.sh
--- perl5.003_07.orig/hints/mpeix.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/mpeix.sh	2008-09-19 18:23:00.000000000 +0100
@@ -68,3 +68,4 @@
 i_termios='undef'
 d_tcgetpgrp='undef'
 d_tcsetpgrp='undef'
+i_db=$undef
diff -ru perl5.003_07.orig/hints/ncr_tower.sh perl5.003_07/hints/ncr_tower.sh
--- perl5.003_07.orig/hints/ncr_tower.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/ncr_tower.sh	2008-09-19 18:23:00.000000000 +0100
@@ -14,3 +14,4 @@
 usemymalloc='y'
 useposix='false'
 so='none'
+i_db=$undef
diff -ru perl5.003_07.orig/hints/netbsd.sh perl5.003_07/hints/netbsd.sh
--- perl5.003_07.orig/hints/netbsd.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/netbsd.sh	2008-09-19 18:23:00.000000000 +0100
@@ -37,3 +37,4 @@
     archname=`uname -m`-${osname}
     ;;
 esac
+i_db=$undef
diff -ru perl5.003_07.orig/hints/next_3_0.sh perl5.003_07/hints/next_3_0.sh
--- perl5.003_07.orig/hints/next_3_0.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/next_3_0.sh	2008-09-19 18:23:00.000000000 +0100
@@ -46,3 +46,4 @@
 # (Thanks to Andreas Koenig <k@franz.ww.tu-berlin.de>)
 ranlib='sleep 5; /bin/ranlib' 
 
+i_db=$undef
diff -ru perl5.003_07.orig/hints/next_3.sh perl5.003_07/hints/next_3.sh
--- perl5.003_07.orig/hints/next_3.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/next_3.sh	2008-09-19 18:23:00.000000000 +0100
@@ -86,3 +86,4 @@
 # This is true whether we're on an HPPA machine or cross-compiling
 # for one.
 pp_cflags='optimize=""'
+i_db=$undef
diff -ru perl5.003_07.orig/hints/next_4.sh perl5.003_07/hints/next_4.sh
--- perl5.003_07.orig/hints/next_4.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/next_4.sh	2008-09-19 18:23:00.000000000 +0100
@@ -101,3 +101,4 @@
 # fails with the -O flag on pp.c.
 # But since there is no HPPA for OPENSTEP...
 # pp_cflags='optimize="-g"'
+i_db=$undef
diff -ru perl5.003_07.orig/hints/opus.sh perl5.003_07/hints/opus.sh
--- perl5.003_07.orig/hints/opus.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/opus.sh	2008-09-19 18:23:00.000000000 +0100
@@ -1 +1,2 @@
 ccflags="$ccflags -X18"
+i_db=$undef
diff -ru perl5.003_07.orig/hints/os2.sh perl5.003_07/hints/os2.sh
--- perl5.003_07.orig/hints/os2.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/os2.sh	2008-09-19 18:23:00.000000000 +0100
@@ -229,3 +229,4 @@
 
 # Now go back
 cd ../../hints
+i_db=$undef
diff -ru perl5.003_07.orig/hints/powerux.sh perl5.003_07/hints/powerux.sh
--- perl5.003_07.orig/hints/powerux.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/powerux.sh	2008-09-19 18:23:00.000000000 +0100
@@ -66,3 +66,4 @@
 usemymalloc='n'
 ssizetype='ssize_t'
 usevfork='false'
+i_db=$undef
diff -ru perl5.003_07.orig/hints/sco_2_3_0.sh perl5.003_07/hints/sco_2_3_0.sh
--- perl5.003_07.orig/hints/sco_2_3_0.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/sco_2_3_0.sh	2008-09-19 18:23:00.000000000 +0100
@@ -1,2 +1,3 @@
 yacc='/usr/bin/yacc -Sm25000'
 i_dirent=undef
+i_db=$undef
diff -ru perl5.003_07.orig/hints/sco_2_3_1.sh perl5.003_07/hints/sco_2_3_1.sh
--- perl5.003_07.orig/hints/sco_2_3_1.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/sco_2_3_1.sh	2008-09-19 18:23:00.000000000 +0100
@@ -1,2 +1,3 @@
 yacc='/usr/bin/yacc -Sm25000'
 i_dirent=undef
+i_db=$undef
diff -ru perl5.003_07.orig/hints/sco_2_3_2.sh perl5.003_07/hints/sco_2_3_2.sh
--- perl5.003_07.orig/hints/sco_2_3_2.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/sco_2_3_2.sh	2008-09-19 18:23:00.000000000 +0100
@@ -1,2 +1,3 @@
 yacc='/usr/bin/yacc -Sm25000'
 libswanted=`echo " $libswanted "| sed 's/ x / /'`
+i_db=$undef
diff -ru perl5.003_07.orig/hints/sco_2_3_3.sh perl5.003_07/hints/sco_2_3_3.sh
--- perl5.003_07.orig/hints/sco_2_3_3.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/sco_2_3_3.sh	2008-09-19 18:23:00.000000000 +0100
@@ -1,3 +1,4 @@
 yacc='/usr/bin/yacc -Sm25000'
 echo "NOTE: you may have problems due to a spurious semicolon on the strerror()"
 echo "macro definition in /usr/include/string.h.  If so, delete the semicolon."
+i_db=$undef
diff -ru perl5.003_07.orig/hints/sco_2_3_4.sh perl5.003_07/hints/sco_2_3_4.sh
--- perl5.003_07.orig/hints/sco_2_3_4.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/sco_2_3_4.sh	2008-09-19 18:23:00.000000000 +0100
@@ -3,3 +3,4 @@
 usemymalloc='y'
 echo "NOTE: you may have problems due to a spurious semicolon on the strerror()"
 echo "macro definition in /usr/include/string.h.  If so, delete the semicolon."
+i_db=$undef
diff -ru perl5.003_07.orig/hints/sco.sh perl5.003_07/hints/sco.sh
--- perl5.003_07.orig/hints/sco.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/sco.sh	2008-09-19 18:23:00.000000000 +0100
@@ -137,3 +137,4 @@
 # in pp_sys.c, but that fails due to a redefinition of struct timeval.
 # This will generate a WHOA THERE.  Accept the default.
 i_sysselct=$undef
+i_db=$undef
diff -ru perl5.003_07.orig/hints/solaris_2.sh perl5.003_07/hints/solaris_2.sh
--- perl5.003_07.orig/hints/solaris_2.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/solaris_2.sh	2008-09-19 18:23:00.000000000 +0100
@@ -348,3 +348,4 @@
 
 End_of_Solaris_Notes
 
+i_db=$undef
diff -ru perl5.003_07.orig/hints/stellar.sh perl5.003_07/hints/stellar.sh
--- perl5.003_07.orig/hints/stellar.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/stellar.sh	2008-09-19 18:23:00.000000000 +0100
@@ -1,2 +1,3 @@
 optimize="-O0"
 ccflags="$ccflags -nw"
+i_db=$undef
diff -ru perl5.003_07.orig/hints/sunos_4_0.sh perl5.003_07/hints/sunos_4_0.sh
--- perl5.003_07.orig/hints/sunos_4_0.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/sunos_4_0.sh	2008-09-19 18:23:00.000000000 +0100
@@ -1,2 +1,3 @@
 ccflags="$ccflags -DFPUTS_BOTCH"
 i_unistd=$undef
+i_db=$undef
diff -ru perl5.003_07.orig/hints/sunos_4_1.sh perl5.003_07/hints/sunos_4_1.sh
--- perl5.003_07.orig/hints/sunos_4_1.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/sunos_4_1.sh	2008-09-19 18:23:00.000000000 +0100
@@ -55,3 +55,4 @@
 # library.
 # 
 # Thanks to William Setzer <William_Setzer@ncsu.edu> for this info.
+i_db=$undef
diff -ru perl5.003_07.orig/hints/svr4.sh perl5.003_07/hints/svr4.sh
--- perl5.003_07.orig/hints/svr4.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/svr4.sh	2008-09-19 18:23:00.000000000 +0100
@@ -41,3 +41,4 @@
 before running make.
 
 EOM
+i_db=$undef
diff -ru perl5.003_07.orig/hints/ti1500.sh perl5.003_07/hints/ti1500.sh
--- perl5.003_07.orig/hints/ti1500.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/ti1500.sh	2008-09-19 18:23:00.000000000 +0100
@@ -1 +1,2 @@
 usemymalloc='n'
+i_db=$undef
diff -ru perl5.003_07.orig/hints/titanos.sh perl5.003_07/hints/titanos.sh
--- perl5.003_07.orig/hints/titanos.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/titanos.sh	2008-09-19 18:23:00.000000000 +0100
@@ -37,3 +37,4 @@
 # Setting libpth shouldn't be needed any more.
 # libpth='/usr/lib /usr/local/lib /lib'
 pth='. /bin /usr/bin /usr/ucb /usr/local/bin /usr/X11/bin /usr/lbin /etc /usr/lib'
+i_db=$undef
diff -ru perl5.003_07.orig/hints/ultrix_4.sh perl5.003_07/hints/ultrix_4.sh
--- perl5.003_07.orig/hints/ultrix_4.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/ultrix_4.sh	2008-09-19 18:23:00.000000000 +0100
@@ -51,3 +51,4 @@
 esac
 
 groupstype='int'
+i_db=$undef
diff -ru perl5.003_07.orig/hints/umips.sh perl5.003_07/hints/umips.sh
--- perl5.003_07.orig/hints/umips.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/umips.sh	2008-09-19 18:23:00.000000000 +0100
@@ -37,3 +37,4 @@
 esac
 
 glibpth="$glibpth /usr/lib/cmplrs/cc"
+i_db=$undef
diff -ru perl5.003_07.orig/hints/unicos.sh perl5.003_07/hints/unicos.sh
--- perl5.003_07.orig/hints/unicos.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/unicos.sh	2008-09-19 18:23:00.000000000 +0100
@@ -7,3 +7,4 @@
 d_setregid='undef'
 d_setreuid='undef'
 
+i_db=$undef
diff -ru perl5.003_07.orig/hints/unisysdynix.sh perl5.003_07/hints/unisysdynix.sh
--- perl5.003_07.orig/hints/unisysdynix.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/unisysdynix.sh	2008-09-19 18:23:00.000000000 +0100
@@ -1 +1,2 @@
 d_waitpid=undef
+i_db=$undef
diff -ru perl5.003_07.orig/hints/utekv.sh perl5.003_07/hints/utekv.sh
--- perl5.003_07.orig/hints/utekv.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/utekv.sh	2008-09-19 18:23:00.000000000 +0100
@@ -10,3 +10,4 @@
 echo "NOTE: You may have to take out makefile dependencies on the files in"
 echo "/usr/include (i.e. /usr/include/ctype.h) or the make will fail.  A"
 echo "simple 'grep -v /usr/include/ makefile' should suffice."
+i_db=$undef
diff -ru perl5.003_07.orig/hints/uts.sh perl5.003_07/hints/uts.sh
--- perl5.003_07.orig/hints/uts.sh	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/hints/uts.sh	2008-09-19 18:23:00.000000000 +0100
@@ -1,2 +1,3 @@
 ccflags="$ccflags -DCRIPPLED_CC"
 d_lstat=define
+i_db=$undef
diff -ru perl5.003_07.orig/makedepend.SH perl5.003_07/makedepend.SH
--- perl5.003_07.orig/makedepend.SH	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/makedepend.SH	2008-09-19 17:53:28.000000000 +0100
@@ -96,7 +96,7 @@
     *.y) filebase=`basename $file .y` ;;
     esac
     case "$file" in
-    */*) finc="-I`echo $file | sed 's#/[^/]*$##`" ;;
+    */*) finc="-I`echo $file | sed 's#/[^/]*$##'`" ;;
     *)   finc= ;;
     esac
     $echo "Finding dependencies for $filebase$obj_ext."
@@ -111,13 +111,21 @@
 	-e '}' ) >UU/$file.c
     $cppstdin $finc -I/usr/local/include -I. $cppflags $cppminus <UU/$file.c |
     $sed \
-	-e '/^#.*<stdin>/d' \
-	-e 's#\.[0-9][0-9]*\.c#'"$file.c#" \
-	-e 's/^[	 ]*#[	 ]*line/#/' \
-	-e '/^# *[0-9][0-9]* *[".\/]/!d' \
-	-e 's/^.*"\(.*\)".*$/'$filebase'\$(OBJ_EXT): \1/' \
-	-e 's/^# *[0-9][0-9]* \(.*\)$/'$filebase'\$(OBJ_EXT): \1/' \
-	-e 's|: \./|: |' \
+        -e '1d' \
+        -e '/^#.*<stdin>/d' \
+        -e '/^#.*<builtin>/d' \
+        -e '/^#.*<built-in>/d' \
+        -e '/^#.*<command line>/d' \
+        -e '/^#.*<command-line>/d' \
+        -e '/^#.*"-"/d' \
+        -e '/^#.*"\/.*\/"/d' \
+        -e '/: file path prefix .* never used$/d' \
+        -e 's#\.[0-9][0-9]*\.c#'"$file.c#" \
+        -e 's/^[	 ]*#[	 ]*line/#/' \
+        -e '/^# *[0-9][0-9]* *[".\/]/!d' \
+        -e 's/^.*"\(.*\)".*$/'$filebase'\$(OBJ_EXT): \1/' \
+        -e 's/^# *[0-9][0-9]* \(.*\)$/'$filebase'\$(OBJ_EXT): \1/' \
+        -e 's|: \./|: |' \
 	-e 's|\.c\.c|.c|' | \
     $uniq | $sort | $uniq >> .deptmp
 done
diff -ru perl5.003_07.orig/perl.h perl5.003_07/perl.h
--- perl5.003_07.orig/perl.h	2008-09-19 17:22:57.000000000 +0100
+++ perl5.003_07/perl.h	2008-09-19 17:53:28.000000000 +0100
@@ -1925,5 +1925,39 @@
 #define printf PerlIO_stdoutf
 #endif
 
+#ifndef PERL_SCRIPT_MODE
+#define PERL_SCRIPT_MODE "r"
+#endif
+
+/* provide some backwards compatibility for XS source from 5.005 */
+#define djSP dSP
+#define AvFILLp(a) AvFILL(a)
+#define dTHR typedef int _thr_dummy
+#define ERRSV GvSV(errgv)
+#define ERRHV GvHV(errgv)
+#define DEFSV GvSV(defgv)
+#define SAVE_DEFSV SAVESPTR(GvSV(defgv))
+
+#ifdef HAS_SEM
+#   include <sys/ipc.h>
+#   include <sys/sem.h>
+#   ifndef HAS_UNION_SEMUN	/* Provide the union semun. */
+    union semun {
+	int val;
+	struct semid_ds *buf;
+	unsigned short *array;
+    };
+#   endif
+#   ifdef USE_SEMCTL_SEMUN
+#       define Semctl(id, num, cmd, semun) semctl(id, num, cmd, semun)
+#   else
+#       ifdef USE_SEMCTL_SEMID_DS
+#           define Semctl(id, num, cmd, semun) semctl(id, num, cmd, semun.buf)
+#       endif
+#   endif
+#   ifndef Semctl	/* Place our bets on the semun horse. */
+#       define Semctl(id, num, cmd, semun) semctl(id, num, cmd, semun)
+#   endif
+#endif
 #endif /* Include guard */
 
